pkg/querier/queryrange/query_range.go

```diff
@@ -198,7 +198,7 @@ func (prometheusCodec) EncodeRequest(ctx context.Context, r Request) (*http.Requ
 	return req.WithContext(ctx), nil
 }
 
-func (prometheusCodec) DecodeResponse(ctx context.Context, r *http.Response) (Response, error) {
+func (prometheusCodec) DecodeResponse(ctx context.Context, r *http.Response, _ Request) (Response, error) {
```

Comment:
I would suggest to add a comment to this function, explaining why we have `_ Request`, otherwise there's the risk we'll clean it up in the future because considered dead code.

Reply:
I added a note, but on the interface, I don't here it matters.